src/model/AbstractPhpStruct.php

```diff
@@ -72,11 +71,12 @@ public function __construct(?string $name = null) {
 	/**
 	 * Sets requried files
 	 *
-	 * @param array $files
+	 * @param array|Set $files
+	 *
 	 * @return $this
 	 */
-	public function setRequiredFiles(array $files) {
-		$this->requiredFiles = new Set($files);
+	public function setRequiredFiles(array|Set $files): self {
```

Comment:
what about? 

```suggestion
	public function setRequiredFiles(iterable $files): self {
```

I checked, that `Set` uses `Iterator` - dunno exactly which is the more broader typehint here (would also upstream this to phootwork then).